Improve/performance accessibility seo

RaheeqMousa opened the pull request.

RaheeqMousa committed:
header html and css

RaheeqMousa committed:
footer html and css

RaheeqMousa committed:
download funcitionallity

RaheeqMousa committed:
animated section inside explore page

RaheeqMousa committed:
Home page HTML & CSS

RaheeqMousa committed:
BugFix-Mobile loader visible

RaheeqMousa committed:
BugFix-Download Feature

RaheeqMousa committed:
clean code, update CSS class naming, improve performance

RaheeqMousa committed:
Improve performance

RaheeqMousa committed:
Improve performance

LoayH98 commented on index.html:
```diff
@@ -1,15 +1,125 @@
+<html lang="en">
+
+<head>
+    <title>PicSnap</title>
```
Support title changing based on what page you are in 
like “Home — PicSnap” ...

LoayH98 commented on index.html:
```diff
@@ -1,15 +1,125 @@
+    <title>PicSnap</title>
+    <meta charset="UTF-8">
+    <meta name="description" content="Free Pictures Website">
+    <meta name="keywords" content="HTML, CSS, JavaScript">
```
deprecated, no value for it in SEO

LoayH98 commented on index.html:
```diff
@@ -1,15 +1,125 @@
+    <link href="assets/CSS/main.css" rel="stylesheet">
+    <link rel="preconnect" href="https://cdnjs.cloudflare.com" crossorigin />
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/7.0.0/css/all.min.css"
+        integrity="..." crossorigin="anonymous" referrerpolicy="no-referrer" />
```
whats the integrity means? and why you put it here?

RaheeqMousa replied:
The integrity attribute uses Subresource Integrity (SRI) to ensure that an external file, such as a script or stylesheet, hasn’t been altered. It contains a cryptographic hash of the file’s expected contents, and the browser checks the downloaded file against this hash. If there’s any mismatch the resource is blocked from loading.

LoayH98 commented on index.html:
```diff
@@ -1,15 +1,125 @@
+    <link
+        href="https://fonts.googleapis.com/css2?family=Amiri:ital,wght@0,400;0,700;1,400;1,700&family=Aref+Ruqaa+Ink:wght@400;700&family=Artifika&display=swap"
+        rel="stylesheet" />
+
```
you are missing favicon in the head 
+og:title, og:description, og:image
search about the use of them

LoayH98 commented on index.html:
```diff
@@ -1,15 +1,125 @@
+        <div class="container">
+            <div class="row">
+                <div>
+                    <img src="assets/Images/logo.png" alt="PicSnap logo" title="PicSnap logo">
```
add height and width

LoayH98 commented on index.html:
```diff
@@ -1,15 +1,125 @@
+                        <li><a href="#">login</a></li>
+                        <li><a href="#">register</a></li>
+                    </ul>
+                </nav>
```
why two navs without labeling?

LoayH98 commented on index.html:
```diff
@@ -1,15 +1,125 @@
+
+        <div class="ScrollContainer">
+            <section class="ImgSection row">
+                <span>Explore the beauty of Nature</span>
```
why a span? 
what about the semantic html that we learnt?
use heading instead

LoayH98 commented on index.html:
```diff
@@ -1,15 +1,125 @@
+                <div>
+                    <img src="assets/Images/reading.jpg" alt="Reading books" title="Hourse" width="300px"
+                        height="300px">
+                    <img src="assets/Images/flowers in nature.jpg" alt="flowers" title="Flower" width="300px"
```
consider adding loading="lazy" to non-critical images,

LoayH98 commented on index.html:
```diff
@@ -1,15 +1,125 @@
+                <nav>
+                    <ul class="row links">
+                        <li>
+                            <a href="https://www.linkedin.com/in/raheeq-mousa" target="_blank"
```
read about `rel="noopener noreferrer"` and check if needed when you put 'a' tags

RaheeqMousa replied:
When using target="_blank" on links, the new page can access the original page via window.opener, which can allow malicious sites to redirect the page or manipulate its content. Adding rel="noopener" prevents this by blocking the new page from accessing window.opener, improving security. 
Including noreferrer also hides the referrer URL, adding a layer of privacy.

LoayH98 commented on explore.html:
```diff
@@ -0,0 +1,142 @@
+<html lang="en">
+
+<head>
+    <title>PicSnap</title>
```
same here PicSnap - Explore

LoayH98 commented on explore.html:
```diff
@@ -0,0 +1,142 @@
+    <meta name="author" content="Raheeq Mousa">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link href="assets/CSS/main.css" rel="stylesheet">
+    <link rel="preconnect" href="https://cdnjs.cloudflare.com" crossorigin />
```
whats the need of this? 
answer down please..

LoayH98 commented on explore.html:
```diff
@@ -0,0 +1,142 @@
+    <title>PicSnap</title>
+    <meta charset="UTF-8">
+    <meta name="description" content="Explore Pictures">
+    <meta name="keywords" content="HTML, CSS, JavaScript">
```
deprecated

LoayH98 commented on explore.html:
```diff
@@ -0,0 +1,142 @@
+                    <img src="assets/Images/logo.png" alt="PicSnap logo" title="PicSnap logo">
+                </div>
+
+                <nav>
```
same here for labeling, search why its needed

LoayH98 commented on explore.html:
```diff
@@ -0,0 +1,142 @@
+        </div>
+
+        <section class="explore row">
+            <span>Step outside your comfort zone and explore the beauty our world!</span>
```
dont use span in wrong way

LoayH98 commented on explore.html:
```diff
@@ -0,0 +1,142 @@
+
+        </section>
+
+        <section class="modal DisplayNoneModal">
```
Accessibility-> add role="dialog", aria-modal="true", and connect a label via aria-labelledby or aria-label.

LoayH98 commented on assets/CSS/base.css:
```diff
@@ -0,0 +1,60 @@
+    max-width:1000px;
+    position: relative;
+    left: 50%;
+    transform: translateX(-50%);
```
why im seeing this again?

LoayH98 commented on assets/CSS/main.css:
```diff
@@ -0,0 +1,413 @@
+
+/*......................Modal...........................*/
+
+.modal{
```
lets make the modal as popup not a full screen 
search how popups looks like

LoayH98 commented on assets/CSS/main.css:
```diff
@@ -0,0 +1,413 @@
+ 
+.ImageModal img{
+    width:100vw;
+    height: 100vh;
```
you already have padding and with this height and width it will always overflow
fix it

LoayH98 commented on assets/JS/main.js:
```diff
@@ -0,0 +1,152 @@
+
+async function getImages() {
+    try {
+        const elements = await fetch(`https://picsum.photos/v2/list?page=${page}`);
```
better to reduce the size of the page to 10, performance wise as I mentioned before

LoayH98 commented on assets/JS/main.js:
```diff
@@ -0,0 +1,152 @@
+            `
+        }
+
+        modal();
```
modal() adds event listeners every time structurePage() runs (on each infinite scroll). This will duplicate handlers and cause multiple executions. Initialize the modal listeners once on page load, not per page fetch.

LoayH98 commented on assets/JS/main.js:
```diff
@@ -0,0 +1,152 @@
+    const scrollPosition = window.scrollY + window.innerHeight;
+    const threshold = document.documentElement.scrollHeight - 100;
+
+    if (!isLoading && scrollPosition >= threshold) {
```
where is the IntersectionObserver use? 
refactor it to use it.

LoayH98 commented on assets/JS/main.js:
```diff
@@ -0,0 +1,152 @@
+ function modal(){
+
+    const modal= document.querySelector('.modal');
+    const imgs = Array.from(document.querySelectorAll(".ImageWrapper img"));
```
clean code: this isn't used right?

LoayH98 commented on assets/JS/main.js:
```diff
@@ -0,0 +1,152 @@
+                    link.download=`${clickedImage.getAttribute('title')||"image"}.jpg`;
+                    link.click();
+
+                    URL.revokeObjectURL(blobTempURL);
```
The download process is fine, but some browsers require the anchor to be in the DOM. If you see failures, append to body, click, then remove.

LoayH98 commented on assets/JS/main.js:
```diff
@@ -0,0 +1,152 @@
+                    URL.revokeObjectURL(blobTempURL);
+                } catch (error) {
+                    console.error("Download failed:", error);
+                    alert("Unable to download image due to cross-origin restrictions \"CORS\".");
```
consider sanitizing the filename, and adding a small toast instead of alert

LoayH98 commented on assets/JS/main.js:
```diff
@@ -0,0 +1,152 @@
+
+    closeBtn.addEventListener("click",function(e){
+        modal.classList.add('DisplayNoneModal');
+    })
```
add Escape key listener and overlay click-to-close.
the overlay click-to-close will be around the modal its called backdrop

LoayH98 reviewed (commented):
Good job, please check the comments, fix what needed, and let me know about the answers of the questions I added - you can add them as replay to the comment of it-

RaheeqMousa committed:
Add favicon, and og:title, og:image, and og:description

RaheeqMousa committed:
Add favicon, and og:title, og:image, and og:description

RaheeqMousa committed:
Enhance modal and UI interactions

- Add overlay backdrop and around-click to close modals
- Add Escape key support for modal closing
- Implement IntersectionObserver for scroll-based loading
- Add toast notifications
- Use rel='noopener noreferrer' for security
- Add aria-label attributes for accessibility

RaheeqMousa committed:
Update main.css

RaheeqMousa committed:
Update main.css

RaheeqMousa committed:
fixed Naming Convention

RaheeqMousa committed:
fixed Naming Convention
